src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.dto.PathBag;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+public class PathFinder {
+    private final WeightedMultigraph<Station, DefaultWeightedEdge> graph;
+
+    private PathFinder(WeightedMultigraph<Station, DefaultWeightedEdge> weightedMultigraph) {
+        this.graph = weightedMultigraph;
+    }
+
+    public static PathFinder from(WeightedMultigraph<Station, DefaultWeightedEdge> weightedMultigraph) {
```

Comment:
```suggestion
    public static PathFinder from(Station) {
```

외부 라이브러리의 종속성 범위 좁혀보면 어떨까요?
추후 라이브러리가 변경되거나, 버전이 변경으로 인한 대응시

많은 부분이 외부라이브러리와 역여있다면 
유지보수하기에 어렵다는 생각이 들어서요!

Reply:
피드백 주신 내용대로 외부 라이브러리에 대한 내용들이 코드 곳곳에 퍼져있으면 
그만큼 외부 변경사항으로 인한 관리 포인트가 늘어날 것 같아요 !
따라서 피드백 주신 내용 반영하여 
외부 라이브러리에 대한 종속성을 `PathFinder` 클래스 내부로 캡슐화 하였습니다
좋은 의견 감사합니다 ! 😄

Reply:
```
public class SectionEdge extends DefaultWeightedEdge {}
public class SubwayGraph extends WeightedMultigraph<Station, SectionEdge> {}
```

이렇게 하는 관점도 있습니다 :smile: 
참고해주세요~

Reply:
피드백 주신 내용대로 외부 라이브러리를 그대로 사용하는 것 말고도
도메인에 맞게 확장해서 사용하는 방법도 있네요 !
당장 기능의 확장이 필요해서 상속을 사용하는 것은 아니지만
도메인의 네이밍으로 라이브러리를 확장한 클래스를 사용함으로써
해당 도메인에서 어떤 기능을 할지 충분히 설명이 가능하고 앞으로 기능의 확장에도 열려있는 구조로 개선이 가능할 것 같아요 !
피드백 주신 내용 수용하여 반영하였습니다
2단계 끝까지 세심하게 봐주셔서 감사합니다 !! 🙇🏻‍♂️👍